clippy_lints/src/methods/iter_count.rs

```diff
@@ -0,0 +1,62 @@
+use crate::utils::{
+    derefs_to_slice, is_type_diagnostic_item, match_type, method_chain_args, paths, snippet_with_applicability,
+    span_lint_and_sugg,
+};
+
+use if_chain::if_chain;
+use rustc_errors::Applicability;
+use rustc_hir::Expr;
+use rustc_lint::LateContext;
+use rustc_span::sym;
+
+use super::ITER_COUNT;
+
+pub(crate) fn lints<'tcx>(cx: &LateContext<'tcx>, expr: &Expr<'_>, iter_args: &'tcx [Expr<'tcx>], is_mut: bool) {
+    let mut_str = if is_mut { "_mut" } else { "" };
+    let iter_method = if method_chain_args(expr, &[format!("iter{}", mut_str).as_str(), "count"]).is_some() {
+        "iter"
+    } else if method_chain_args(expr, &["into_iter", "count"]).is_some() {
+        "into_iter"
+    } else {
+        return;
+    };
```

Comment:
Are these lines needed? I guess receiving a iter-method name as an arg seems to be enough, but what about?

Reply:
I've just modified `lints()` in order to receive iter method name as an argument.

Reply:
It looks good. Thanks!